### PR TITLE
Add gNMI path to change Stratum log level

### PR DESCRIPTION
**Pull request by pudelkoM**
`/system/logging/console/config/severity` can be used to set the glog level.

Example:

```bash
bazel run --config=asan //stratum/tools/gnmi:gnmi_cli -- --grpc_addr=10.128.13.27:9339 set --string_val="DEBUG" '/system/logging/console/config/severity'
```

TODO:

- [x] Add comment/docs about this new feature
- [x] Unit tests

### CODE REVIEW

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 1]
```diff
@@ -3205,6 +3205,104 @@ void SetUpComponentsComponentIntegratedCircuitStateNodeId(uint64 node_id,
...
+      return SendResponse(
+          GetResponse(path, ConvertLogSeverityToString(minloglevel, verbosity)),
+          stream);
+    };
+    node->SetOnTimerHandler(poll_functor)->SetOnPollHandler(poll_functor);
+
+    // TODO(max): do we need this?
+    // Trigger change notification.
+    // tree->SendNotification(GnmiEventPtr(
+    //     new PortHealthIndicatorChangedEvent(node_id, port_id, typed_state)));
```
Comment (bocon13) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc:3265]: I don't know who is going to listen to this. Let's wait to implement the on change stuff until someone requests it in the future.

File: stratum/hal/lib/common/yang_parse_tree_paths.cc [hunk 2]
```diff
@@ -3205,6 +3205,104 @@ void SetUpComponentsComponentIntegratedCircuitStateNodeId(uint64 node_id,
...
+    return SendResponse(
+        GetResponse(path, ConvertLogSeverityToString(minloglevel, verbosity)),
+        stream);
+  };
+
+  // TODO(max): do we need this?
+  // auto register_functor = RegisterFunc<OpticalOutputPowerChangedEvent>();
+  // auto on_change_functor = GetOnChangeFunctor(
+  //     module, network_interface, &OpticalOutputPowerChangedEvent::GetMax,
+  //     &ConvertDoubleToDecimal64OrDie);
```
Comment (bocon13) [inline stratum/hal/lib/common/yang_parse_tree_paths.cc:3297]: I don't think we need this yet.

Comment (pudelkoM) [reply stratum/hal/lib/common/yang_parse_tree_paths.cc:3297]: I think this is related to the other place above. Sending the notification allows subscribing `on-change` here?

Comment (pudelkoM) [reply stratum/hal/lib/common/yang_parse_tree_paths.cc:3297]: I confirmed the above. Without the events `onchange` subs get nothing.

File: stratum/hal/lib/common/gnmi_events.h [hunk 3]
```diff
@@ -442,6 +461,7 @@ class ConfigHasBeenPushedEvent
       : new_config_(new_config) {}
   ~ConfigHasBeenPushedEvent() override {}
 
+  //  private:
```
Comment (pudelkoM) [inline stratum/hal/lib/common/gnmi_events.h:464]: remove

**Review by Yi-Tseng — DISMISSED** [review]

File: stratum/hal/lib/common/yang_parse_tree_test.cc [hunk 4]
```diff
@@ -4692,5 +4702,303 @@ TEST_F(YangParseTreeOpticalChannelTest,
...
+
+  // Check reaction to wrong value type.
+  ::gnmi::Value wrong_type_val;
+  EXPECT_THAT(ExecuteOnUpdate(path, wrong_type_val,
+                              /* SetValue will not be called */ nullptr,
+                              /* Notification will not be called */ nullptr),
+              StatusIs(_, _, ContainsRegex("not a TypedValue message")));
+}
+
+// Check if the '/interfaces/interface/config/health-indicator' OnReplace action
```
Comment (pudelkoM) [inline stratum/hal/lib/common/yang_parse_tree_test.cc:4961]: fix

File: stratum/hal/lib/common/yang_parse_tree_test.cc [hunk 5]
```diff
@@ -4692,5 +4702,303 @@ TEST_F(YangParseTreeOpticalChannelTest,
...
+                      ConsoleLogSeverityChangedEvent(kGlogSeverityFatalString,
+                                                     kGlogVerbosityZeroString),
+                      &resp));
+  // Check that the result of the call is what is expected.
+  ASSERT_THAT(resp.update().update(), SizeIs(1));
+  EXPECT_EQ(resp.update().update(0).val().string_val(),
+            kSeverityCriticalString);
+}
+
+// Check if the '/system/logging/console/config/severity' OnPoll action works
```
Comment (pudelkoM) [inline stratum/hal/lib/common/yang_parse_tree_test.cc:4814]: fix

File: stratum/hal/lib/common/yang_parse_tree_test.cc [hunk 6]
```diff
@@ -4692,5 +4702,303 @@ TEST_F(YangParseTreeOpticalChannelTest,
...
+                      ConsoleLogSeverityChangedEvent(kGlogSeverityFatalString,
+                                                     kGlogVerbosityZeroString),
+                      &resp));
+  // Check that the result of the call is what is expected.
+  ASSERT_THAT(resp.update().update(), SizeIs(1));
+  EXPECT_EQ(resp.update().update(0).val().string_val(),
+            kSeverityCriticalString);
+}
+
+// Check if the '/interfaces/interface/config/health-indicator' OnUpdate action
```
Comment (pudelkoM) [inline stratum/hal/lib/common/yang_parse_tree_test.cc:4920]: fix

**Pull request merged**